[docs] What's the lab about? (#19611)
mui-org/material-ui: docs/src/pages/components/about-the-lab/about-the-lab.md

<p class="description">This package hosts the incubator components that are not yet ready to move to the core.</p>

> The main difference between the lab and the core is how the components are versioned. Having the lab within Material UI allows us to release breaking changes when necessary while the core package follows a [slower-moving policy](https://material-ui.com/versions/#release-frequency).

Reviewer: ```suggestion
The main difference between the lab and the core is how the components are versioned. Having the lab within Material-UI allows us to release breaking changes when necessary while the core package follows a [slower-moving policy](https://material-ui.com/versions/#release-frequency).
```

> The main difference between the lab and the core is how the components are versioned. Having the lab within Material UI allows us to release breaking changes when necessary while the core package follows a [slower-moving policy](https://material-ui.com/versions/#release-frequency).

Reviewer 2: ```suggestion
The main difference between the lab and the core is how the components are versioned. Having a separate lab package allows us to release breaking changes when necessary while the core package follows a [slower-moving policy](https://material-ui.com/versions/#release-frequency).
```

> As our users use and test the components and report issues, we learn more about the design flaws of our components: lacking features, a11y, bugs, API, etc. The older and more used a component is, the less likely it is that we will find new flaws and subsequently, need to introduce breaking changes.

Reviewer: I wonder, shouldn't we remove the usage of "our"/"us"? Here is how it would look like in this paragraph.

```suggestion
As developers use and test the components and report issues, the maintainers learn more about the design flaws of the components: lacking features, accessibility, bugs, API, etc. The older and more used a component is, the less likely it is that new flaws will be found and subsequently, need to introduce breaking changes.
```

> As our users use and test the components and report issues, we learn more about the design flaws of our components: lacking features, a11y, bugs, API, etc. The older and more used a component is, the less likely it is that we will find new flaws and subsequently, need to introduce breaking changes.

Reviewer 2: ```suggestion
As developers use and test the components and report issues, the maintainers learn more about shortcomings of the components: missing features, accessibility issues, bugs, API design, etc. The older and more used a component is, the less likely it is that new issues will be found and subsequently need to introduce breaking changes.
```

> For a component to be ready to move to the core, the following criteria is considered:

Reviewer 2: ```suggestion
For a component to be ready to move to the core, the following criteria are considered:
```

* It needs to be **used**. The Material UI team uses Google Analytics stats to evaluate the usage of each component. A lab component with low usage either means that it isn't fully working yet or that there is a low demand for it.

> * It needs to match the **code quality** of the core components. It doesn't have to be perfect to be a part of the core. We want the component to be reliable enough so that developers can depend on our solution.

Reviewer 2: ```suggestion
* It needs to match the **code quality** of the core components. It doesn't have to be perfect to be a part of the core, but the component should be reliable enough that developers can depend on it.
```

>     * Each component needs **type definitions**. It currently is not required that a lab component is typed, but it would need to be typed to move to the core.

Reviewer 2: ```suggestion
    * Each component needs **type definitions**. It is not currently required that a lab component is typed, but it would need to be typed to move to the core.
```

    * Requires good **test coverage**. Some of the lab components don't currently have comprehensive tests.
* Can it be used as **leverage** to incentivize users to upgrade to the latest major release? The less fragmented the community is, the better.

> * It needs to have a low probability of a **breaking change** in the short/medium future. For instance, if we know we want to introduce a new feature that will likely require a breaking change, we might as well delay its promotion to the core.

Reviewer 2: ```suggestion
* It needs to have a low probability of a **breaking change** in the short/medium future. For instance, if it needs a new feature that will likely require a breaking change, it may be preferable to delay its promotion to the core.
```
